conveyor/worker.go

```diff
@@ -0,0 +1,525 @@
+	w.postponedResponses = append(w.postponedResponses, w.slot.responseQueue.RemoveAll()...)
+	w.nextWorkerState = ProcessElements
+
+	totalNumElements := len(w.postponedResponses)
+	numProcessedElements := 0
+	for i := 0; i < totalNumElements; i++ {
+		resp := w.postponedResponses[i]
+		if resp.GetItemType() > 9999 { // TODO: check isNestedEvent
+			w.processNestedEvent(resp)
+		} else {
+			err := w.processResponse(resp)
+			if err != nil {
+				return errors.Wrap(err, "[ readResponseQueue ] Can't processResponse")
+			}
+		}
+
+		numProcessedElements++
+
+		if w.slot.inputQueue.HasSignal() {
+			w.nextWorkerState = ReadInputQueue
+			break
+		}
+	}
+
+	w.postponedResponses = w.postponedResponses[numProcessedElements:]
+
+	return nil
+}
+
+func (w *worker) waitQueuesOrTick() {
+	log.Debugf("[ waitQueuesOrTick ] starts ... ( pulseState: %s )", w.slot.pulseState.String())
+	time.Sleep(time.Millisecond * 300)
+	//panic("[ waitQueuesOrTick ] implement me") // TODO :
+}
+
+func (w *worker) processingElements() {
+	log.Debugf("[ processingElements ] starts ... ( pulseState: %s )", w.slot.pulseState.String())
+	if !w.slot.hasElements(ActiveElement) {
+		if w.slot.pulseState == constant.Past {
+			if w.slot.hasExpired() {
+				w.slot.slotState = Suspending
+				log.Info("[ processingElements ] Set slot state to 'Suspending'")
+				return
+			}
+		}
+		w.waitQueuesOrTick()
+	}
+
+	if w.slot.inputQueue.HasSignal() {
+		log.Info("[ processingElements ] Set next worker state to 'ReadInputQueue'")
+		w.nextWorkerState = ReadInputQueue
+		return
+	}
+
+	numActiveElements := w.slot.len(ActiveElement)
+	breakProcessing := false
+	for ; numActiveElements > 0 && !breakProcessing; numActiveElements-- {
+		element := w.slot.popElement(ActiveElement)
+		lastState := -1
+		for lastState < int(element.state) {
+			lastState = int(element.state)
+
+			if w.processOneElement(element) {
+				break
+			}
+
+			if w.slot.inputQueue.HasSignal() {
+				w.nextWorkerState = ReadInputQueue
+				log.Info("[ processingElements ] Set next worker state to 'ReadInputQueue'")
+				breakProcessing = true
+				break
+			}
+		}
+
+		err := w.slot.pushElement(element)
+		if err != nil {
+			panic(fmt.Sprintf("[ processingElements ] Can't push element: %+v", element))
+		}
+	}
+}
+
+func (w *worker) processOneElement(element *slotElement) bool {
+	transitionHandler := element.stateMachineType.GetTransitionHandler(w.slot.pulseState, element.state)
+	payLoad, newState, err := transitionHandler(element)
+	if err != nil {
+		log.Error("[ processingElements ] Transition handler error: ", err)
+		errorHandler := element.stateMachineType.GetTransitionErrorHandler(w.slot.pulseState, element.state)
+		payLoad, newState = errorHandler(element, err)
+	}
+	setNewElementState(element, payLoad, newState)
+
+	stopProcessingElement := newState == 0
+
+	return stopProcessingElement
+}
+
+func (w *worker) working() {
```

Comment:
What about switch-case by `w.nextWorkerState` inside `for`? We will need `ReadResponseQueue` state than, and set it in` w.readInputQueueWorking()`